File: src/org/nextras/orm/intellij/completion/ModifiersProvider.java
package org.nextras.orm.intellij.completion;

import com.intellij.codeInsight.completion.*;
import com.intellij.codeInsight.lookup.LookupElement;
import com.intellij.codeInsight.lookup.LookupElementBuilder;
import com.intellij.util.ProcessingContext;
import com.jetbrains.php.PhpIcons;
import org.jetbrains.annotations.NotNull;


public class ModifiersProvider extends CompletionProvider<CompletionParameters>
{
	public static ModifierInsertHandler withParams = new ModifierInsertHandler(true);
	public static ModifierInsertHandler withoutParams = new ModifierInsertHandler(false);

	@Override
	protected void addCompletions(@NotNull CompletionParameters params,
	                              ProcessingContext context,
	                              @NotNull CompletionResultSet result)
	{
		result.addElement(LookupElementBuilder.create("enum").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("default").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("virtual").withInsertHandler(withoutParams));
		result.addElement(LookupElementBuilder.create("container").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("1:m").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("m:1").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("1:1").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("1:1d").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("m:n").withInsertHandler(withParams));
		result.addElement(LookupElementBuilder.create("primary").withInsertHandler(withoutParams));
	}

}

File: src/main/kotlin/org/nextras/orm/intellij/completion/ModifiersProvider.kt
package org.nextras.orm.intellij.completion

import com.intellij.codeInsight.completion.CompletionParameters
import com.intellij.codeInsight.completion.CompletionProvider
import com.intellij.codeInsight.completion.CompletionResultSet
import com.intellij.codeInsight.lookup.LookupElementBuilder
import com.intellij.util.ProcessingContext

class ModifiersProvider : CompletionProvider<CompletionParameters>() {
	override fun addCompletions(
		params: CompletionParameters,
		context: ProcessingContext,
		result: CompletionResultSet
	) {
		result.addElement(LookupElementBuilder.create("enum").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("default").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("virtual").withInsertHandler(withoutParams))
		result.addElement(LookupElementBuilder.create("1:m").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("m:1").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("1:1").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("m:m").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("primary").withInsertHandler(withoutParams))
		result.addElement(LookupElementBuilder.create("primary-proxy").withInsertHandler(withoutParams))
		result.addElement(LookupElementBuilder.create("wrapper").withInsertHandler(withParams))
		result.addElement(LookupElementBuilder.create("embeddable").withInsertHandler(withoutParams))
	}

	companion object {
		private val withParams = ModifierInsertHandler(true)
		private val withoutParams = ModifierInsertHandler(false)
	}
}
